app/lib/cloneutils.js

```diff
@@ -46,7 +46,10 @@ export const clone = (repoURL, destination, progressCallback, token) => {
     }
 
     if (token) {
-      options.fetchOpts.callbacks.credentials = function () {
+      options.fetchOpts.callbacks.certificate_check = () => { // for libgit2 problem
```

Comment:
Configuration for nodegit to use access token. Will probably switch to electron bindings for git in the future for better performance

Reply:
I'd love to see `// for libgit2 problem` be elaborated on to understand why we've currently skipping this check, for future reference

Reply:
Just added some documentation in the code to better explain the problem, here's a link to the issue I put in code: https://github.com/nodegit/nodegit/tree/master/guides/cloning/gh-two-factor#github-certificate-issue-in-os-x